test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1121,53 +1079,205 @@ UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup, compare_cm_and_wireup_configs,
+    connect_and_fail_wireup(sender(), FAIL_WIREUP_MSG_SEND,
+                            UCP_EP_FLAG_CONNECT_REQ_QUEUED);
 }
 
-UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup,
+UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup_fail,
                      connect_and_fail_wireup_msg_send_on_server,
                      !cm_use_all_devices())
 {
-    connect_and_fail_wireup(receiver(), UCP_EP_FLAG_CONNECT_PRE_REQ_QUEUED,
-                            FAIL_WIREUP_MSG_SEND);
+    connect_and_fail_wireup(receiver(), FAIL_WIREUP_MSG_SEND,
+                            UCP_EP_FLAG_CONNECT_PRE_REQ_QUEUED);
 }
 
-UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup,
+UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup_fail,
                      connect_and_fail_wireup_msg_pack_addr_on_client,
                      !cm_use_all_devices())
 {
-    connect_and_fail_wireup(sender(), UCP_EP_FLAG_CLIENT_CONNECT_CB,
-                            FAIL_WIREUP_MSG_ADDR_PACK);
+    connect_and_fail_wireup(sender(), FAIL_WIREUP_MSG_ADDR_PACK,
+                            UCP_EP_FLAG_CLIENT_CONNECT_CB);
 }
 
-UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup,
+UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup_fail,
                      connect_and_fail_wireup_msg_pack_addr_on_server,
                      !cm_use_all_devices())
 {
-    connect_and_fail_wireup(receiver(), UCP_EP_FLAG_SERVER_NOTIFY_CB,
-                            FAIL_WIREUP_MSG_ADDR_PACK);
+    connect_and_fail_wireup(receiver(), FAIL_WIREUP_MSG_ADDR_PACK,
+                            UCP_EP_FLAG_SERVER_NOTIFY_CB);
 }
 
-UCS_TEST_P(test_ucp_sockaddr_wireup,
+UCS_TEST_P(test_ucp_sockaddr_wireup_fail,
            connect_and_fail_wireup_set_ep_failed_on_client)
 {
-    connect_and_fail_wireup(sender(), UCP_EP_FLAG_CLIENT_CONNECT_CB,
-                            FAIL_WIREUP_SET_EP_FAILED);
+    connect_and_fail_wireup(sender(), FAIL_WIREUP_SET_EP_FAILED,
+                            UCP_EP_FLAG_CLIENT_CONNECT_CB);
 }
 
-UCS_TEST_P(test_ucp_sockaddr_wireup,
+UCS_TEST_P(test_ucp_sockaddr_wireup_fail,
            connect_and_fail_wireup_set_ep_failed_on_server)
 {
-    connect_and_fail_wireup(receiver(), UCP_EP_FLAG_SERVER_NOTIFY_CB,
-                            FAIL_WIREUP_SET_EP_FAILED);
+    connect_and_fail_wireup(receiver(), FAIL_WIREUP_SET_EP_FAILED,
+                            UCP_EP_FLAG_SERVER_NOTIFY_CB);
 }
 
-UCP_INSTANTIATE_ALL_TEST_CASE(test_ucp_sockaddr_wireup)
+UCP_INSTANTIATE_ALL_TEST_CASE(test_ucp_sockaddr_wireup_fail)
+
+
+class test_ucp_sockaddr_wireup_fail_try_next_cm :
+        public test_ucp_sockaddr_wireup_fail  {
+public:
+    test_ucp_sockaddr_wireup_fail_try_next_cm()
+    {
+        m_try_next_cm_cb_found = false;
+    }
+
+private:
+    static int find_try_next_cm_cb(const ucs_callbackq_elem_t *elem, void *arg)
+    {
+        if ((elem->cb == ucp_cm_client_try_next_cm_progress) &&
+            (elem->arg == arg)) {
+            m_try_next_cm_cb_found = true;
+        }
```

Comment:
maybe check that there is no callback where elem->arg==arg?
it would be more generic

Reply:
but it could have some other callbacks

Reply:
i think no callback should remain when ep is destroyed?

Reply:
> i think no callback should remain when ep is destroyed?

yes, but here we are waiting for the `ucp_cm_client_try_next_cm_progress` being scheduled on a progress - which means that CM started fallback procedure. and we won't to emulate error at this moment to check that EP removes this callback from queue

Reply:
if the test is only checking ucp_cm_client_try_next_cm_progress is scheduled, what would make it fail if it is not removed?
can u pls add this as a comment in the test code?

Reply:
> if the test is only checking ucp_cm_client_try_next_cm_progress is scheduled, what would make it fail if it is not removed?

```
[1626089685.130340] [jazz04:63475:0]       callbackq.c:452  UCX  WARN  0 fast-path and 1 slow-path callbacks remain in the queue
common/test.cc:353: Failure
Failed
Got 0 errors and 1 warnings during the test
```

Reply:
nice

Reply:
> can u pls add this as a comment in the test code?

done